File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/V2TableMigrator.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.base.destination.typing_deduping;

/**
 * Prefer {@link io.airbyte.integrations.base.destination.typing_deduping.migrators.Migration}
 * instead.
 */
public interface V2TableMigrator {

  void migrateIfNecessary(final StreamConfig streamConfig) throws Exception;

}

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/V2TableMigrator.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.base.destination.typing_deduping

/**
 * Prefer [io.airbyte.integrations.base.destination.typing_deduping.migrators.Migration] instead.
 */
interface V2TableMigrator {
    @Throws(Exception::class) fun migrateIfNecessary(streamConfig: StreamConfig?)
}
